src/App.tsx

```diff
@@ -28,11 +29,27 @@ type ReorderOptions = {
 
 // Use this function in the render method to prepare goods
 export function getReorderedGoods(
-  goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+  let visibleGoods = [...goodsFromServer];
+
+  if (sortType === SortType.ALPHABET) {
+    visibleGoods = visibleGoods.sort((a, b) => a.localeCompare(b));
+  } else if (sortType === SortType.LENGTH) {
+    visibleGoods = visibleGoods.sort((a, b) => {
+      const lengthDiffrence = a.length - b.length;
+
+      if (lengthDiffrence === 0) {
+        return a.localeCompare(b);
+      }
+
+      return lengthDiffrence;
+    });
+  }
```

Comment:
1. I agree with Matylda, it can be const and then you shouldn't assign value to it, just use the method to modify it
2. you can simplify your function using ifs inside sort method

```suggestion
  const visibleGoods = [...goodsFromServer];

  visibleGoods.sort((goodA, goodB) => {
    switch (sortType) {
      case SortType.ALPHABET:
        return goodA.localeCompare(goodB);
      case SortType.LENGTH:
        return goodA.length - goodB.length;
      default:
        return 0;
    }
  });

  if (isReversed) {
    visibleGoods.reverse();
  }
```